Building my first extension missing test result in failure

### Describe the bug

running:

`mvn io.quarkus.platform:quarkus-maven-plugin:3.5.2:create-extension -N -DgroupId=io.quarkiverse.code-server -DextensionId=code-server -DwithoutTests`

as documented in https://quarkus.io/guides/building-my-first-extension results in the project build failing:

```
mvn install
Using maven at '/Users/manderse/.sdkman/candidates/mvnd/current/bin/mvnd' to run buildFile '/Users/manderse/code/quarkus-code-server/pom.xml':
Nov 20, 2023 1:09:00 PM org.jline.utils.Log logr
WARNING: Unable to create a system terminal, creating a dumb terminal (enable debug logging for more information)
[main] WARN org.jline - Unable to create a system terminal, creating a dumb terminal (enable debug logging for more information)
[INFO] Processing build on daemon 6aea3359
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] Child module /Users/manderse/code/quarkus-code-server/integration-tests of /Users/manderse/code/quarkus-code-server/pom.xml does not exist @
 @
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]
[ERROR]   The project io.quarkiverse.code-server:quarkus-code-server-parent:999-SNAPSHOT (/Users/manderse/code/quarkus-code-server/pom.xml) has 1 error
[ERROR]     Child module /Users/manderse/code/quarkus-code-server/integration-tests of /Users/manderse/code/quarkus-code-server/pom.xml does not exist
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
```



### Expected behavior

that getting started docs does not suggest create project that does not build :)

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/quarkiverse/java/pom.tpl.qute.xml
+++ b/independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/quarkiverse/java/pom.tpl.qute.xml
@@ -21,6 +21,7 @@
         </pluginManagement>
     </build>
     {/if}
+    {#if has-integration-tests-module}
     <profiles>
         <profile>
             <id>it</id>
@@ -35,4 +36,5 @@
             </modules>
         </profile>
     </profiles>
+    {/if}
 </project>

--- a/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/codestarts/extension/QuarkusExtensionCodestartCatalog.java
+++ b/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/codestarts/extension/QuarkusExtensionCodestartCatalog.java
@@ -50,7 +50,8 @@ public final class QuarkusExtensionCodestartCatalog extends GenericCodestartCata
         IT_PARENT_RELATIVE_PATH("it-parent.relative-path"),
         MAVEN_QUARKUS_EXTENSION_PLUGIN("maven.quarkus-extension-plugin"),
         MAVEN_COMPILER_PLUGIN_VERSION("maven.compiler-plugin-version"),
-        HAS_DOCS_MODULE("has-docs-module");
+        HAS_DOCS_MODULE("has-docs-module"),
+        HAS_INTEGRATION_TESTS_MODULE("has-integration-tests-module");
 
         private final String key;
 

--- a/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/CreateExtension.java
+++ b/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/CreateExtension.java
@@ -8,6 +8,7 @@ import static io.quarkus.devtools.codestarts.extension.QuarkusExtensionCodestart
 import static io.quarkus.devtools.codestarts.extension.QuarkusExtensionCodestartCatalog.QuarkusExtensionData.EXTENSION_NAME;
 import static io.quarkus.devtools.codestarts.extension.QuarkusExtensionCodestartCatalog.QuarkusExtensionData.GROUP_ID;
 import static io.quarkus.devtools.codestarts.extension.QuarkusExtensionCodestartCatalog.QuarkusExtensionData.HAS_DOCS_MODULE;
+import static io.quarkus.devtools.codestarts.extension.QuarkusExtensionCodestartCatalog.QuarkusExtensionData.HAS_INTEGRATION_TESTS_MODULE;
 import static io.quarkus.devtools.codestarts.extension.QuarkusExtensionCodestartCatalog.QuarkusExtensionData.IT_PARENT_ARTIFACT_ID;
 import static io.quarkus.devtools.codestarts.extension.QuarkusExtensionCodestartCatalog.QuarkusExtensionData.IT_PARENT_GROUP_ID;
 import static io.quarkus.devtools.codestarts.extension.QuarkusExtensionCodestartCatalog.QuarkusExtensionData.IT_PARENT_RELATIVE_PATH;
@@ -105,6 +106,7 @@ public class CreateExtension {
     private String extensionsRelativeDir = "extensions";
     private boolean withCodestart;
     private String javaVersion;
+    private boolean hasIntegrationTestsModule;
 
     public CreateExtension(final Path baseDir) {
         this.baseDir = requireNonNull(baseDir, "extensionDirPath is required");
@@ -218,6 +220,7 @@ public class CreateExtension {
     }
 
     public CreateExtension withoutIntegrationTests(boolean withoutIntegrationTest) {
+        hasIntegrationTestsModule = !withoutIntegrationTest;
         this.builder.withoutIntegrationTests(withoutIntegrationTest);
         return this;
     }
@@ -263,6 +266,7 @@ public class CreateExtension {
         data.putIfAbsent(CLASS_NAME_BASE, toCapCamelCase(extensionId));
         data.put(EXTENSION_FULL_NAME,
                 data.getRequiredStringValue(NAMESPACE_NAME) + data.getRequiredStringValue(EXTENSION_NAME));
+        data.put(HAS_INTEGRATION_TESTS_MODULE, hasIntegrationTestsModule);
 
         // for now, we only support Java extensions
         data.put(JAVA_VERSION, javaVersion == null ? JavaVersion.DEFAULT_JAVA_VERSION_FOR_EXTENSION

